### PR TITLE
Allow to retry tempest run

### PR DESCRIPTION
**Pull request by cjeanner**
From time to time, tempest fails due to OpenStack API going down. Let's
allow it to re-run after a small delay to see if it can avoid a full
"recheck" in RDO.

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running
- [X] Appropriate documentation exists and/or is up-to-date:
  - [X] README in the role

### CODE REVIEW

**Comment by pablintino** [comment]
Is this the only way of solving this?

**Comment by cjeanner** [comment]
> Is this the only way of solving this?

if you have any better idea... ? Maybe we can lower the strength of the hammer, by lowering the concurrency (defaults to 4) ?

**Comment by pablintino** [comment]
> > Is this the only way of solving this?
> 
> if you have any better idea... ? Maybe we can lower the strength of the hammer, by lowering the concurrency (defaults to 4) ?

We would need to investigate what are the usual services to go down, and check if they have limits set. Which ones are  usually affected?

**Comment by cjeanner** [comment]
> > > Is this the only way of solving this?
> > 
> > 
> > if you have any better idea... ? Maybe we can lower the strength of the hammer, by lowering the concurrency (defaults to 4) ?
> 
> We would need to investigate what are the usual services to go down, and check if they have limits set. Which ones are usually affected?

Not sure if this is something caught in the dashboards, but that's probably of interest. Maybe something to discuss with @adrianfusco and @arxcruz . We can't realistically manually check that specific data.

File: ci_framework/roles/tempest/tasks/main.yml
Comment (kopecmartin) [inline ci_framework/roles/tempest/tasks/main.yml:40]: nit: Wrap

**Comment by kopecmartin** [comment]
This is definitely a bug, API can't just go down, I'd report it and link it from the code stating that the retry is a workaround until the bug is fixed. If all API is down, it's probably OpenShift bug and if only a few services' API are down, that it's a bug of those services ...

**Comment by lewisdenny** [comment]
IMO we should add this retry for now with a `todo` commented in the code to remove it once Tempest is stable and create a Jira to track this issue if we don't already have one.

**Review by lewisdenny — APPROVED** [review]
Seems lowing concurrency didn't help, I think we should merge this to unblock the jobs tempest failures. 

I've opened a Jira ticket to track going forward.

**Comment by TristanCacqueray** [comment]
Hello, I'm using these build results to fine-tune [logreduce](https://github.com/logreduce/logreduce) for ci-framework jobs, and here are some odd events discovered in https://review.rdoproject.org/zuul/build/eadeea960cc34f96b81796980e11c014:
- Thousands of `:2379: connection refused` in [kube-apiserver](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openshift-kube-apiserver_kube-apiserver-crc-74q6p-master-0_f3f6deb5bdd034d233ae6bdeb750d116/kube-apiserver/)
- `dial tcp 10.217.4.1:443: connect: no route to host` in [ingress_router](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openshift-ingress_router-default-7d9f7cf6ff-kxt2q_63b18229-1f25-4983-aadc-0f315455a8d8/router/1.log)

- In rabbitmq-server-cell*-server logs they all say that `Node database directory at /var/lib/rabbitmq/mnesia/rabbit@rabbitmq-cell1-server-1.rabbitmq-cell1-nodes.openstack is empty. Assuming we need to join an existing cluster or initialise from scratch...`, which was not the case in previous builds. ([server-0](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openstack_rabbitmq-cell1-server-0_9e92c2db-a340-4c13-8d20-2179e42d07e1/rabbitmq/0.log), [server-1](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openstack_rabbitmq-cell1-server-1_c608f3ab-8a8e-4503-9eac-7104bd738d39/rabbitmq/0.log) and [server-2](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openstack_rabbitmq-cell1-server-2_69f65502-e7ed-4b0e-8b19-55a2912728df/rabbitmq/0.log)).
- Similarly in galera appears to emit new warning such as `WSREP: Member 2.0 (openstack-galera-1) requested state transfer from '*any*', but it is impossible to select State Transfer donor: Resource temporarily unavailable`: [galera-2](https://logserver.rdoproject.org/98/598/34027528031deed140efd98e2f6c16c045c619b2/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/eadeea9/controller/ci-framework-data/logs/crc/pods/openstack_openstack-galera-2_02316699-53c9-408e-8d60-2d775034c100/galera/0.log)

Hope that helps!

**Comment by cjeanner** [comment]
/hold

We removed galera replicas via https://github.com/openstack-k8s-operators/ci-framework/pull/615 - let's check if it's making things better without that horrible hack.

**Comment by cjeanner** [comment]
not sure this is really still needed. Let's close, and see how things go in the future.

**Pull request closed without merging**
